Xamarin.Forms.Platform.Android/CollectionView/ObservableItemsSource.cs

```diff
@@ -23,6 +22,28 @@ public ObservableItemsSource(IEnumerable itemSource, RecyclerView.Adapter adapte
 
 		public object this[int index] => _itemsSource[index];
 
+		public void Dispose()
+		{
+			Dispose(true);
+		}
+
+		bool _disposed;
+
+		protected virtual void Dispose(bool disposing)
+		{
+			if (!_disposed)
+			{
+				if (disposing)
+				{
+					((INotifyCollectionChanged)_itemsSource).CollectionChanged -= CollectionChanged;
+					_itemsSource = null;
+					_adapter = null;
```

Comment:
No point in setting `_adapter` and `_itemsSource` to `null` here, this instance is going away anyway.